model/patch_lifecycle.go

```diff
@@ -738,6 +738,7 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string) (*Vers
 			},
 		)
 	}
+	SetNumDependents(tasksToInsert)
```

Comment:
Because the `CreateBuildFromVersionNoInsert` calls that occur inside the loop no longer set the tasks num dependents field, we now need to do it right before we [insert all tasks for the version](https://github.com/evergreen-ci/evergreen/blob/6249b9d4bc06adfab7ba965446b45b770cfd7a4c/model/patch_lifecycle.go#L781)